stdlib/public/runtime/Errors.cpp

```diff
@@ -22,11 +22,80 @@
 #include <pthread.h>
 #include <stdarg.h>
 #include "swift/Runtime/Debug.h"
+#include "swift/Basic/Demangle.h"
+#include "swift/Config.h"
+#include <regex>
+#include <cxxabi.h>
+#include <iostream>
+#include <execinfo.h>
 
 #ifdef __APPLE__
 #include <asl.h>
 #endif
 
+static std::string
+cppDemangle(std::string line, bool *success)
+{
+  std::regex cppRegex("(.*?)(_Z[_a-zA-Z0-9$]+)(.*)");
```

Comment:
This will have false positives when _Z… appears in the middle of a symbol name. You should match a non-identifier character before _Z.